### ISSUE TITLE
Inconsistent segment parameter requirements (Trac #2135)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2127]
`I3Tray.AddSegment` assumes that segments take two parameters (the tray and a name), but `icetray.traysegment` enforces only that they must have one parameter (the tray):

```text
@icetray.traysegment
def mySegment(tray):
	pass
tray=I3Tray()
tray.AddSegment(mySegment)
```

results in:

```text
  ...
  File "/Users/cweaver/Work/IceCube/combo/build/lib/I3Tray.py", line 224, in AddSegment
    return _segment(self, _name, **kwargs)
TypeError: mySegment() takes exactly 1 argument (2 given)
```

This is not caught by [http://code.icecube.wisc.edu/projects/icecube/browser/IceTray/projects/icetray/trunk/python/traysegment.py#L23 the relevant check in icetray.traysegment]. 

Migrated from https://code.icecube.wisc.edu/ticket/2135
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T21:03:41", 
    "description": "`I3Tray.AddSegment` assumes that segments take two parameters (the tray and a name), but `icetray.traysegment` enforces only that they must have one parameter (the tray):\n\n{{{\n@icetray.traysegment\ndef mySegment(tray):\n\tpass\ntray=I3Tray()\ntray.AddSegment(mySegment)\n}}}\n\nresults in:\n\n{{{\n  ...\n  File \"/Users/cweaver/Work/IceCube/combo/build/lib/I3Tray.py\", line 224, in AddSegment\n    return _segment(self, _name, **kwargs)\nTypeError: mySegment() takes exactly 1 argument (2 given)\n}}}\n\nThis is not caught by [http://code.icecube.wisc.edu/projects/icecube/browser/IceTray/projects/icetray/trunk/python/traysegment.py#L23 the relevant check in icetray.traysegment]. ", 
    "reporter": "cweaver", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547240621058514", 
    "component": "combo core", 
    "summary": "Inconsistent segment parameter requirements", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-02-12T23:26:32", 
    "milestone": "", 
    "owner": "cweaver", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180213T15:51:34: 
* cweaver changed owner from "" to "cweaver"
* cweaver commented:

> In changeset 3030:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="3030"
> Change traysegment to enforce the rules used by I3Tray.AddSegment.
> 
> Fixes #2127
> ```


* cweaver changed resolution from "" to "fixed"
* cweaver changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T21:03:41: nwhitehorn commented:

> In changeset 168851:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168851"
> Change traysegment to enforce the rules used by I3Tray.AddSegment.
> 
> Fixes #2127
> ```
